feature/platform/tests/optics_power_and_bias_current_test/optics_power_and_bias_current_test.go

```diff
@@ -0,0 +1,179 @@
+			if len(outputPowers) == 0 {
+				t.Errorf("Get biasCurrents list for %q: got 0, want > 0", transceiver)
+			}
+		}
+	}
+}
+
+func TestOpticsPowerUpdate(t *testing.T) {
+	dut := ondatra.DUT(t, "dut")
+	dp := dut.Port(t, "port1")
+	d := &telemetry.Device{}
+	i := d.GetOrCreateInterface(dp.Name())
+
+	cases := []struct {
+		desc                string
+		IntfStatus          bool
+		expectedStatus      telemetry.E_Interface_OperStatus
+		expectedMaxOutPower float64
+		checkMinOutPower    bool
+	}{{
+		// Check both input and output optics power are in normal range.
+		desc:                "Check initial input and output optics powers are OK",
+		IntfStatus:          true,
+		expectedStatus:      telemetry.Interface_OperStatus_UP,
+		expectedMaxOutPower: maxOpticsPower,
+		checkMinOutPower:    true,
+	}, {
+		desc:                "Check output optics power is very small after interface is disabled",
+		IntfStatus:          false,
+		expectedStatus:      telemetry.Interface_OperStatus_DOWN,
+		expectedMaxOutPower: minOpticsPower,
+		checkMinOutPower:    false,
+	}, {
+		desc:                "Check output optics power is normal after interface is re-enabled",
+		IntfStatus:          true,
+		expectedStatus:      telemetry.Interface_OperStatus_UP,
+		expectedMaxOutPower: maxOpticsPower,
+		checkMinOutPower:    true,
+	}}
+	for _, tc := range cases {
+		t.Log(tc.desc)
+		intUpdateTime := 2 * time.Minute
+		t.Run(tc.desc, func(t *testing.T) {
+			i.Enabled = ygot.Bool(tc.IntfStatus)
+			dut.Config().Interface(dp.Name()).Replace(t, i)
+			dut.Telemetry().Interface(dp.Name()).OperStatus().Await(t, intUpdateTime, tc.expectedStatus)
+
+			// TODO: Remove t.Skipf() after the issue is fixed.
+			t.Skipf("Optics from INNOLIGHT is not supported, skip it for now.")
+
+			channels := dut.Telemetry().Component(dp.Name()).Transceiver().ChannelAny()
+			inputPowers := channels.InputPower().Instant().Get(t)
+			outputPowers := channels.OutputPower().Instant().Get(t)
+			for _, inPower := range inputPowers {
+				if inPower > maxOpticsPower || inPower < minOpticsPower {
+					t.Errorf("Get inputPower for port %q): got %.2f, want within [%f, %f]", dp.Name(), inPower, minOpticsPower, maxOpticsPower)
+				}
+			}
+			for _, outPower := range outputPowers {
+				if outPower > tc.expectedMaxOutPower {
+					t.Errorf("Get outPower for port %q): got %.2f, want < %f", dp.Name(), outPower, tc.expectedMaxOutPower)
+				}
+				if tc.checkMinOutPower && outPower < minOpticsPower {
+					t.Errorf("Get outPower for port %q): got %.2f, want > %f", dp.Name(), outPower, minOpticsPower)
+				}
+			}
+		})
+	}
+}
+
+func findComponentsByType(t *testing.T, dut *ondatra.DUTDevice, cType telemetry.E_PlatformTypes_OPENCONFIG_HARDWARE_COMPONENT) []string {
+	components := dut.Telemetry().ComponentAny().Name().Get(t)
+	var s []string
+	for _, c := range components {
+		lookupType := dut.Telemetry().Component(c).Type().Lookup(t)
+		if !lookupType.IsPresent() {
```

Comment:
Prefer to reduce the levels of nesting by eschewing else.

```
for _, c := range components {
  ...
  if !lookupType.IsPresent() {
    ...
    continue
  }
  componentType := lookupType.Val(t)
  ...
}
```

Reply:
done

Reply:
done